src/v/kafka/server/handlers/fetch.cc

```diff
@@ -193,38 +193,34 @@ static ss::future<read_result> do_read_from_ntp(
           error_code::not_leader_for_partition);
     }
 
-    auto kafka_partition = make_partition_proxy(
-      ntp_config.materialized_ntp, partition, mgr);
-    if (!kafka_partition) {
-        return ss::make_ready_future<read_result>(
-          error_code::unknown_topic_or_partition);
-    }
+    auto kafka_partition = make_partition_proxy<replicated_partition>(
+      partition);
 
     if (config::shard_local_cfg().enable_transactions.value()) {
         if (
           ntp_config.cfg.isolation_level
           == model::isolation_level::read_committed) {
-            ntp_config.cfg.max_offset = kafka_partition->last_stable_offset();
+            ntp_config.cfg.max_offset = kafka_partition.last_stable_offset();
```

Comment:
will this commit compile ? `kafka_partition` is still `std::optional<parititon_proxy>` ?

Reply:
Its a bit confusing, but this compiles. At this point in the commit history there are two methods called `make_partition_proxy`.
```
template<typename Impl, typename... Args>
partition_proxy make_partition_proxy(Args&&... args) {
    return partition_proxy(std::make_unique<Impl>(std::forward<Args>(args)...));
}

std::optional<partition_proxy> make_partition_proxy(
  const model::ntp&,
  ss::lw_shared_ptr<cluster::partition>,
  cluster::partition_manager&);
```
The change above this comment calls the other one, which returns a `replicated_partition`.